app/src/main/kotlin/compiler/intermediate/generators/DefaultGeneratorDetailsGenerator.kt

```diff
@@ -0,0 +1,250 @@
+package compiler.intermediate.generators
+
+import compiler.ast.AstNode
+import compiler.ast.Function
+import compiler.ast.NamedNode
+import compiler.ast.Statement
+import compiler.ast.Type
+import compiler.ast.Variable
+import compiler.intermediate.CFGLinkType
+import compiler.intermediate.ControlFlowGraph
+import compiler.intermediate.ControlFlowGraphBuilder
+import compiler.intermediate.IFTNode
+import compiler.intermediate.Register
+import compiler.intermediate.SummedConstant
+import compiler.utils.Ref
+
+class DefaultGeneratorDetailsGenerator(
+    private val parameters: List<NamedNode>,
+    initLabel: IFTNode.MemoryLabel,
+    private val resumeLabel: IFTNode.MemoryLabel,
+    finalizeLabel: IFTNode.MemoryLabel,
+    depth: ULong,
+    variablesLocationTypes: Map<Ref<NamedNode>, VariableLocationType>,
+    displayAddress: IFTNode,
+    private val nestedForeachLoops: List<Ref<Statement.ForeachLoop>>,
+    private val nameResolution: Map<Ref<AstNode>, Ref<NamedNode>>, // constructor doesn't use this argument, thus incomplete map can be passed
```

Comment:
These two maps are only used in one place. Maybe it would be easier to pass instead a function that for a given foreach loop in `nestedForeachLoops` returns `GeneratorDetailsGenerator`